src/components/profile/index.js

```diff
@@ -10,41 +13,131 @@ import InputField from 'components/common/input-field'
 import ActionButton from 'components/common/action-button'
 import ROUTE from 'constants/route'
 import EditButton from 'components/common/edit-button'
+import STYLE from 'constants/style'
 
 import style from './index.style.js'
 
 export type PropsType = {
   authError: Object,
   css?: Object,
   children?: React.Node,
-  email?: string,
+  storeId: IdType,
+  store: Object,
+  storeName: string,
   password?: string,
+  walletAddress: string,
   signOut: () => Promise<any>,
   handleChange: {},
-} & ReactRouter.ContextRouter
+  onClick: () => void,
+  onUpdateAddress: ({
+    walletAddress: string,
+    storeName: string,
+    storeId: IdType,
+  }) => void,
+  ...ReactRouter.ContextRouter,
+}
 
 type StateType = {
-  email?: string,
-  password?: string,
+  isEditingEmail: boolean,
+  isEditingPassword: boolean,
+  isEditingAddress: boolean,
+  addressField: string,
 }
 
 class Profile extends React.Component<PropsType, StateType> {
   state = {
-    email: 'test@test.com',
-    password: 'password',
+    isEditingAddress: false,
+    isEditingPassword: false,
+    isEditingEmail: false,
+    addressField: this.props.walletAddress,
+  }
+
+  //////////////////////
+  // LIFECYCLE EVENTS //
+  //////////////////////
+
+  componentDidMount() {
+    const addressField = this.getWalletAddress()
+    // debugger
+    console.log('componentWillMount_addressField: ', addressField)
+    this.setState({ addressField })
+  }
+
+  componentDidUpdate() {
+    console.log('state.isEditingAddress: ', this.state.isEditingAddress)
+  }
+
+  ////////////////////
+  // EVENT HANDLERS //
+  ////////////////////
+
+  handleLogOut = async () => {
+    await this.props.signOut()
+    this.props.history.push(ROUTE.PATH.HOME)
   }
 
+  handleChange = (e: SyntheticEvent<HTMLButtonElement>) => {
+    // e.persist()
+    e.preventDefault()
+
+    // console.log('e.currentTarget.id', e.currentTarget.id)
+    if (!e.currentTarget.value) {
+      return
+    }
+
+    this.setState(
+      {
+        addressField: e.currentTarget.value,
+      },
+      () => {
+        console.log('this.state:', this.state)
+      }
+    )
+  }
+
+  onClick = (e: SyntheticEvent<HTMLButtonElement>) => {
+    const eventId = e.currentTarget.id
+    this.setState(prevState => {
+      return { [eventId]: !prevState[eventId] }
+    })
+  }
+
+  handleUpdateAddress = async () => {
+    // console.log('this.state:', this.state)
```

Comment:
Make an `ethUtil.isValidAddress(address?: string): boolean` and use it here before submitting. Show error in the same way we do it during sign up

Reply:
👌 used ethUtil.isValidAddress from ethereumjs-util package.